test/tests/criticality/material_density/tests

```diff
@@ -8,6 +8,14 @@
     requirement = 'The system shall error if a criticality search is not paired with the correct problem class.'
     capabilities = 'openmc'
   []
+  [negative_keff_target]
```

Comment:
For a test case, would you please add one that uses the `keff_target` feature and confirms the criticality search behaves as expected?